tree/dataframe/src/RSqliteDS.cxx

```diff
@@ -13,19 +13,304 @@
     \ingroup dataframe
     \brief RDataFrame data source class for reading SQlite files.
 */
+
 // clang-format on
 
-#include <ROOT/RSqliteDS.hxx>
-#include <ROOT/RDFUtils.hxx>
-#include <ROOT/RMakeUnique.hxx>
+#include "ROOT/RSqliteDS.hxx"
+
+#include "ROOT/RConfig.h"
+#include "ROOT/RDFUtils.hxx"
+#include "ROOT/RMakeUnique.hxx"
 
-#include <TError.h>
+#include "TError.h"
+#include "TRandom.h"
+#include "TSystem.h"
 
 #include <algorithm>
 #include <cctype>
 #include <cstring>
+#include <ctime>
+#include <memory> // for placement new
 #include <stdexcept>
 
+#ifdef R__HAS_DAVIX
+#include <davix.hpp>
+#endif
+
+namespace {
+
+constexpr char const *gSQliteVfsName = "ROOT-Davix-readonly";
+
+#ifdef R__HAS_DAVIX
+
+/**
+ * The "derived class" from sqlite3_file that gets passed to the file operation callbacks.
+ */
+struct VfsRootFile {
```

Comment:
we need a long-ish explanation of how this inheritance with no inheritance work :) c++ developers won't know